features/step_definitions/journey_test_steps.rb

```diff
@@ -47,6 +68,10 @@
   click_link_or_button(button_name)
 end
 
+When /I click the '(.*)' button for the supplier user '(.*)'$/ do |button_name,user_name|
+  find(:xpath, "//*/span[contains(text(),'#{user_name}')]/../../td/*//button[contains(text(),'#{button_name}')]").click
```

Comment:
This works OK, but is pretty convoluted XPATH.  I can't see a better way to do things with the way the app is at the moment, but maybe we could make a change to the admin app itself to make the right button for a user more easily findable.

For example, we might be able to add a 'name' attribute to each button (e.g. name='Activate user foo@bar.com'), which would mean the tests can just look for that button directly rather than finding the row with the username and navigating from there through the DOM.

@quis - what do you think?  Am I suggesting something really awful?  I appreciate it's not good to add attributes and bloat the code solely to make tests work, but a descriptive `name` (or something) on a button might not be the end of the world, and would make this considerably less brittle?